evalml/pipelines/time_series.py

```diff
@@ -0,0 +1,82 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.regression_pipeline import RegressionPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import any_values_are_nan, drop_nan, pad_with_nans
+
+
+class TimeSeriesRegressionPipeline(RegressionPipeline):
+    problem_type = ProblemTypes.TIME_SERIES_REGRESSION
+
+    def __init__(self, parameters, gap, max_delay, random_state=0):
+        super().__init__(parameters, random_state)
+        self.gap = gap
+        self.max_delay = max_delay
+
+    def fit(self, X, y=None):
+        """Fit a time series regression pipeline.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray): The target training targets of length [n_samples]
+
+        Returns:
+            self
+        """
+        if y is None:
+            assert isinstance(X, pd.Series), "When modeling a single time series, the data must be a series."
+        if not isinstance(X, pd.DataFrame):
+            X = pd.DataFrame(X)
+
+        if y is None:
+            X_t, _ = self._compute_features_during_fit(X, y=None)
+            y = X.squeeze()
```

Comment:
Shouldn't `y` just remain `None` if it wasn't provided to `fit`?

Reply:
This is for the case when there are no features and only the y is used. In #1396 we decided to use X=None for that and I just pushed that change up! So it should hopefully be clearer now.